File: app/src/main/java/com/mehul/redditwall/favorites/FavViewModel.java
package com.mehul.redditwall.favorites;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import java.util.List;

public class FavViewModel extends AndroidViewModel {
    private FavRepository repo;
    private LiveData<List<FavImage>> fav;

    public FavViewModel(@NonNull Application application) {
        super(application);
        repo = new FavRepository(application.getApplicationContext());
        fav = repo.getAllFav();
    }

    public LiveData<List<FavImage>> getAllFav() {
        return fav;
    }

    public List<FavImage> getFavList() {
        return repo.getFavAsList();
    }

    public void deleteAll() {
        repo.deleteAll();
    }

    public void insert(FavImage saved) {
        repo.insert(saved);
    }

    public void deleteFavImage(FavImage saved) {
        repo.deleteFav(saved);
    }
}

File: app/src/main/java/com/mehul/redditwall/favorites/FavViewModel.kt
package com.mehul.redditwall.favorites

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData

public class FavViewModel(application: Application) : AndroidViewModel(application) {
    private val repo: FavRepository = FavRepository(application.applicationContext)
    val allFav: LiveData<List<FavImage?>?>?

    val favList: List<FavImage?>?
        get() = repo.favAsList

    fun deleteAll() {
        repo.deleteAll()
    }

    fun insert(saved: FavImage?) {
        repo.insert(saved)
    }

    fun deleteFavImage(saved: FavImage?) {
        repo.deleteFav(saved)
    }

    init {
        allFav = repo.allFav
    }
}